### PR TITLE
Reorder v5.2 release notes to showcase more changes

### PR DESCRIPTION
**Pull request by thibaudcolas**
Changes the v5.2 release notes to have additional top-level sections:

- `ModelViewSet` improvements
- User interface refinements
- xternal links in promoted search results
- Under Maintenance – Stimulus adoption

Also includes fixing a few typos

### CODE REVIEW

File: docs/releases/5.2.md [hunk 1]
```diff
@@ -41,55 +41,73 @@ Wagtail now officially supports client-side admin customisations with [Stimulus]
...
+ * Enable reference index tracking for models registered with `ModelViewSet` (Sage Abdullah)
+
+In addition, the following new features have been added to the generic admin views as part of `ModelViewSet`, which can also be used with `SnippetViewSet`.
+
+ * Allow overriding `IndexView.export_headings` via `ModelViewSet` (Christer Jensen, Sage Abdullah)
+ * Add the ability to define listing buttons on generic `IndexView` (Sage Abdullah)
+
+### User interface refinements
+
+Here are tweaks to the admin user interface which we hope will make it easier to use.
```
Comment (gasman) [inline docs/releases/5.2.md:71]: "Here are tweaks" reads a bit oddly - how about
```suggestion
Several tweaks have been made to the admin user interface which we hope will make it easier to use.
```

File: docs/releases/5.2.md [hunk 2]
```diff
@@ -41,55 +41,73 @@ Wagtail now officially supports client-side admin customisations with [Stimulus]
...
  * Add [`linked_fields` mechanism on chooser widgets](chooser_viewsets_limiting_choices) to allow choices to be limited by fields on the calling page (Matt Westcott)
  * Add support for merging cells within `TableBlock` with the [`mergedCells` option](table_block_options) (Gareth Palmer)
  * When adding a panel within `InlinePanel`, focus will now shift to that content similar to `StreamField` (Faishal Manzar)
- * Show the full first published at date within a tooltip on the Page status sidebar on the relative date (Rohit Sharma)
- * Extract generic breadcrumbs functionality from page breadcrumbs (Sage Abdullah)
  * Add support for `placement` in `human_readable_date` the tooltip template tag (Rohit Sharma)
- * Add breadcrumbs support to custom `ModelViewSet` views (Sage Abdullah)
  * Support passing extra context variables via the `{% component %}` tag (Matt Westcott)
  * Allow subclasses of `PagesAPIViewSet` override default [Page model via the `model`](api_v2_configure_endpoints) attribute (Neeraj Yetheendran, Herbert Poul)
  * Allow `ModelViewSet` to be used with models that have non-integer primary keys (Sage Abdullah)
```
Comment (laymonage) [inline docs/releases/5.2.md:101]: This has been moved above

```suggestion
```

**Review by laymonage — APPROVED** [review]
Looks good to me, will merge with the suggested changes. Thanks!

**Comment by laymonage** [comment]
- `main`: 4da28a4
- `stable/5.2.x`: 93c0dd057b2628fc0023e5e4f66f580d53a217a6

**Pull request merged**